### ISSUE TITLE
Notice: Deprecation

### ISSUE THREAD

**Issue by paragonie-scott** [paragonie/password_lock#7]
https://github.com/paragonie/halite/blob/master/src/Password.php

Use Halite instead.

**Comment by sstok** [comment]
Halite is GNU GPL, is it possible to use this with MIT licensed products?
Using Composer to install, and not actually providing both within the same package distribution.

**Comment by paragonie-scott** [comment]
Officially, Halite is available under two licenses:

* GPL3 as you mentioned
* MIT + an invoice

I think our choice is generally fair: The technical aspects of GPL don't matter as much with PHP as they do with complied languages like C, since you always receive the code anyway. All it does is prevent the use of snakeoil code obfuscation techniques.

However, large companies' legal teams are generally allergic to GPL. Additionally, it's easier for them to get a tax write-off for a business expense than it is for donations. And since Paragon Initiative Enterprises is a for-profit company, donations would be unnecessarily hairy anyway.

Large companies have also largely been a "free rider" on the backs of open source software. This licensing structure allows us to actually capture some of the $ they save in liability and their opportunity costs for avoiding an investment into their own R&D.

This, in turn, allows us to keep a roof over our heads while we do all the free work we've done for various open source projects. That is why we've moved towards GPL or MIT+invoice.

[Reference material on commercial licensing for OSS](http://www.kalzumeus.com/2015/01/28/design-and-implementation-of-csvexcel-upload-for-saas/#oss).

So that's the background. As for your question:

> Halite is GNU GPL, is it possible to use this with MIT licensed products?

**YES.** These are the nuances involved in the answer:

1. If someone has purchased a commercial license from us, then obviously they can.
2. If their project is open source, then also yes ([according to the FSF](https://www.gnu.org/licenses/license-list.en.html#Expat)). However, anyone who uses said project needs to know that, without satisfying condition 1, they can't just use Halite under MIT terms (deploy their app on their user's hardware without making the source code available). Our opinion (as the copyright holder) is that linking to the Halite Github repository satisfies this requirement since Halite would still be made available to their users.

I hope that satisfactorily answers your question as well as any curiosity about why we went the way we did.

> Using Composer to install, and not actually providing both within the same package distribution.

I'm not sure what you mean?
